src/components/ModalQueuePriorizations.vue

```diff
@@ -0,0 +1,226 @@
+<template>
+  <UnnnicModal
+    class="queue-modal"
+    :text="$t('chats.select_services_queues')"
+    @close="$emit('close')"
+  >
+    <section class="queue-modal-form">
+      <div
+        v-if="!verifySelectedLength"
+        class="queue-modal-disclaimer"
+      >
+        <UnnnicIconSvg
+          filled
+          icon="alert-circle-1"
+          size="md"
+          scheme="feedback-yellow"
+        />
+        <p>{{ $t('chats.select_at_least') }}</p>
+      </div>
+      <div class="queue-modal-select">
+        <div class="queue-modal-input">
+          <UnnnicLabel :label="$t('chats.select_the_queues')" />
+          <UnnnicSelectSmart
+            v-model="permissionQueues"
+            multiple
+            :options="permissionQueuesOptions"
+            :multipleWithoutSelectsMessage="$t('chats.no_queue_selected')"
+          />
+        </div>
+      </div>
+    </section>
+    <template #options>
+      <UnnnicButton
+        :text="$t('cancel')"
+        type="tertiary"
+        size="large"
+        @click="$emit('close')"
+      />
+      <UnnnicButton
+        :text="$t('save')"
+        type="primary"
+        size="large"
+        :disabled="!verifySelectedLength"
+        @click="saveListQueues"
+      />
+    </template>
+  </UnnnicModal>
+</template>
+<script>
+import { mapState } from 'vuex';
+import callUnnnicAlert from '@/utils/callUnnnicAlert';
+import Queues from '@/services/api/resources/chats/queues';
+
+export default {
+  name: 'ModalQueuePriorizations',
+
+  data() {
+    return {
+      permissionQueues: [],
+      permissionQueuesOptions: [],
+      showModalQueue: false,
+      noQueueSelected: false,
+    };
+  },
+
+  watch: {
+    permissionQueues: {
+      handler: 'updateQueuesPlaceholder',
+      deep: true,
+    },
+  },
+
+  created() {
+    this.getListQueues();
+  },
+
+  computed: {
+    ...mapState({
+      me: (state) => state.profile.me,
+    }),
+
+    verifySelectedLength() {
+      return this.permissionQueues.length > 0;
+    },
+  },
+  methods: {
+    async getListQueues() {
+      try {
+        let me = this.me.email;
+        const response = await Queues.getListQueues(me);
+        response.user_permissions.forEach((permission) => {
+          if (permission.role === 1) {
```

Comment:
I saw that here it also validates for `role === 1`... Instead of declaring several `const ROLE_ID_SELECTED = 1` what do you think about creating a variable in `data` to be unified and accessible throughout the code?

Reply:
Indeed, this is much better